awswrangler/distributed/ray/datasources/arrow_parquet_base_datasource.py

```diff
@@ -0,0 +1,87 @@
+"""Ray ParquetBaseDatasource Module.
+
+This module is pulled from Ray's [ParquetBaseDatasource]
+(https://github.com/ray-project/ray/blob/master/python/ray/data/datasource/parquet_base_datasource.py) with a few changes
+and customized to ensure compatibility with AWS SDK for pandas behavior. Changes from the original implementation,
+are documented in the comments and marked with (AWS SDK for pandas) prefix.
+"""
+
+from typing import Any, Dict, List, Optional
+
+# fs required to implicitly trigger S3 subsystem initialization
+import pyarrow as pa
+import pyarrow.fs
+import pyarrow.parquet as pq
+from ray.data.block import BlockAccessor
+
+from awswrangler._arrow import _add_table_partitions, _df_to_table
+from awswrangler.distributed.ray.datasources.pandas_file_based_datasource import PandasFileBasedDatasource
+
+
+class ArrowParquetBaseDatasource(PandasFileBasedDatasource):  # pylint: disable=abstract-method
```

Comment:
Can you remind me why we need to inherit from `PandasFileBasedDatasource` and not just `FileBasedDatasource`?

Reply:
`PandasFileBasedDatasource` doesn't really have anything specific to Pandas, it just opens our own file system when it comes to writing data. The reason I need to have it here is that `ArrowParquetDatasource` contains the logic for writing Parquet files, which it inherits from `ArrowParquetBaseDatasource`, which in turn inherits it from `PandasFileBasedDatasource`.